com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -889,15 +1038,33 @@ public override void OnLostOwnership()
 
         private void Initialize()
         {
-            ResetInterpolatedStateToCurrentAuthoritativeState(); // useful for late joining
+            if (!IsSpawned)
```

Comment:
wait, how can this happen? If this is called in OnNetworkSpawn, lost and gained ownership, this should never happen, no? Should this throw in that case? (better fail loud and early than silently?)

Reply:
This can happen here:
[NetworkObject.InvokeBehaviourNetworkSpawn](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/blob/e138a9c5612e02acf90bfcfbd381aab9ab5b94b3/com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs#L809)

Reply:
Wait so you can have an ownership callback while not spawned? This seems like a bug no? Shouldn't Ownership callbacks be safe spaces and always online? Does it make sense, from a user's perspective, to have an ownership change while offline?

Reply:
When spawning, you need to determine ownership before you spawn so you know who the owner is when OnNetworkSpawn is invoked.

Reply:
It isn't really "not spawned" but it isn't really "spawned"... it is the quasi state while in the middle of spawning.  It is just an order of operations thing... users want to know who the owner is when OnNetworkSpawn is invoked...so before that is invoked it has to set ownership before invoking OnNetworkSpawn when in the middle of spawning.

Make sense?

Reply:
I agree that internal ownership book keeping should happen before spawn, but should the user side callback be called at that time too? Couldn't internal ownership changes be made, spawn called, then gained ownership callback called?

Reply:
In [NetworkSpawnManager.SpawnNetworkObjectLocallyCommon](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/blob/e138a9c5612e02acf90bfcfbd381aab9ab5b94b3/com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs#L552) both client and server invoke this method at the same place.  

It is a global "order of operations" thing that we can't change as it would change core functionality.
This really isn't that big of a deal if you think about it:

During Spawn:
Ownership initialization is skipped, but OnNetworkSpawn initialization is not.  Since they both call the same method that invokes the same logic, it "just works". 👍 

During Ownership Changes When Spawned:
Ownership initialization is not skipped